File: WebApp - Angular2/app/exposure/position.ts
export interface Position {
  key: string;
  security: string;
  custodian: string;
  executingBroker: string;
  purchaseDate?: Date;
  displayPurchaseDate: string;

  targetAmount: number;
  stagedAmount: number;
  committedAmount: number;
  doneAmount: number;
  leavesAmount: number;
}

File: TickingViewSvc/Models/Position.cs
﻿using System;

namespace TickingViewSvc.Models
{
    public class Position
    {
        public string Key => $"{Security}-{Custodian}-{ExecutingBroker}-{DisplayPurchaseDate}";

        public string Security { get; set; }
        public string Custodian { get; set; }

        public string ExecutingBroker { get; set; }
        public DateTime? PurchaseDate { get; set; }
        public string DisplayPurchaseDate => PurchaseDate.HasValue ? PurchaseDate.Value.ToString("yyyy MMMM dd") : "Today";

        public decimal TargetAmount { get; set; }
        public decimal StagedAmount { get; set; }
        public decimal CommittedAmount { get; set; }
        public decimal DoneAmount { get; set; }
        public decimal LeavesAmount => TargetAmount - DoneAmount;
    }
}
